src/libYARP_dev/src/devices/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -826,8 +826,11 @@ bool ControlBoardWrapper::detachAll()
             yarp::os::RateThread::stop();
 
         int devices=device.subdevices.size();
-        for(int k=0;k<devices;k++)
-            device.getSubdevice(k)->detach();
+        SubDevice* sub;
+        for(int k=0;k<devices;k++) {
+            sub = device.getSubdevice(k);
+            if (sub) sub->detach();
+        }
```

Comment:
Not a big issue, but since the SubDevice is not used outside this `for` loop, my preference would be to keep its life as short as possible, i.e.
```c++
        for(int k=0;k<devices;k++) {
            Subdevice* sub = device.getSubdevice(k);
            if (sub) sub->detach();
        }
```
I don't think that this would impact the performances with any modern compiler, I made a few tests with https://gcc.godbolt.org/ and the assembly generated (with -O0, I could not write a simple test that was not reduced to just a return with -O1 or higher) is exactly the same.

What do other people think about this?

Reply:
I prefer to reduce the scope of the `sub` variable as much as possible. In my limiter experience, `POD` values created in the stack are never a performance problem.

Reply:
Ok, commit updated.